application/src/main/java/org/togetherjava/tjbot/features/basic/SuggestionsUpDownVoter.java

```diff
@@ -55,19 +55,33 @@ public void onMessageReceived(MessageReceivedEvent event) {
     }
 
     private static void createThread(Message message) {
-        String title = message.getContentRaw();
+        String threadTitle = generateThreadTitle(message);
+        message.createThreadChannel(threadTitle).queue();
+    }
 
-        if (title.length() >= TITLE_MAX_LENGTH) {
-            int lastWordEnd = title.lastIndexOf(' ', TITLE_MAX_LENGTH);
+    /**
+     * Generates a title for the given message. The maximum length of the title is
+     * {@value #THREAD_TITLE_MAX_LENGTH}.
+     *
+     * @param message The message for which to generate the title.
+     * @return The generated and truncated thread title.
+     */
+    private static String generateThreadTitle(Message message) {
+        String primaryTitle = message.getContentRaw();
+        String fallbackTitle = message.getAuthor().getName() + "'s suggestion";
```

Comment:
this will use their username rather than nickname, i think using their nickname is a better option